subproject/simple/simple_dragonbox.h

```diff
@@ -0,0 +1,1996 @@
+            b = r < UINT64_C(184467440737095517);
+            s = s * 2 + b;
+            significand = b ? r : significand;
+
+            r = rotr64(uint64_t(significand * UINT64_C(14757395258967641293)), 1);
+            b = r < UINT64_C(1844674407370955162);
+            s = s * 2 + b;
+            significand = b ? r : significand;
+
+            exponent += s;
+        }
+
+        static compute_mul_result<uint64_t> compute_mul(uint64_t u, uint128 cache) {
+            auto const r = umul192_upper128(u, cache);
+            return {r.high, r.low == 0};
+        }
+
+        static uint64_t compute_delta(uint128 cache, int beta) {
+            return cache.high >> (total_bits - 1 - beta);
+        }
+
+        static compute_mul_parity_result compute_mul_parity(uint64_t two_f, uint128 cache, int beta) {
+            assert(beta >= 1);
+            assert(beta < 64);
+            auto const r = umul192_lower128(two_f, cache);
+            return {((r.high >> (64 - beta)) & 1) != 0,
+                    (((r.high << beta) & UINT64_C(0xffffffffffffffff)) | (r.low >> (64 - beta))) == 0};
+        }
+
+        static uint64_t compute_left_endpoint_for_shorter_interval_case(uint128 cache, int beta) {
+            return (cache.high - (cache.high >> (significand_bits + 2))) >>
+                   (total_bits - significand_bits - 1 - beta);
+        }
+
+        static uint64_t compute_right_endpoint_for_shorter_interval_case(uint128 cache, int beta) {
+            return (cache.high + (cache.high >> (significand_bits + 1))) >>
+                   (total_bits - significand_bits - 1 - beta);
+        }
+
+        static uint64_t compute_round_up_for_shorter_interval_case(uint128 cache, int beta) {
+            return ((cache.high >> (total_bits - significand_bits - 2 - beta)) + 1) / 2;
+        }
+
+        template <int N, uint64_t n_max>
+        static uint64_t divide_by_pow10(uint64_t n) {
+            static_assert(N >= 0, "");
+
+            // Specialize for 64-bit division by 10.
+            // Without the bound on n_max (which compilers these days never leverage), the
+            // minimum needed amount of shift is larger than 64.
+            if constexpr (N == 1 && n_max <= UINT64_C(4611686018427387908)) {
+                return umul128_upper64(n, UINT64_C(1844674407370955162));
+            }
+            // Specialize for 64-bit division by 1000.
+            // Without the bound on n_max (which compilers these days never leverage), the
+            // smallest magic number for this computation does not fit into 64-bits.
+            else if constexpr (N == 3 && n_max <= UINT64_C(15534100272597517998)) {
+                return umul128_upper64(n, UINT64_C(4722366482869645214)) >> 8;
+            }
+            else {
+                return n / compute_power<N>(uint64_t(10));
+            }
+        }
+    };
+
+    static constexpr uint64_t cache32[78]{
```

Comment:
I thought you collected every format-specific specializations into `float_format` specializations. If that was indeed what you intended, these also need to go there.